src/AI/Generators/TemplateGenerator.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BumbleDocGen\AI\Generators;
+
+use BumbleDocGen\AI\ProviderInterface;
+use BumbleDocGen\Core\Configuration\Exception\InvalidConfigurationParameterException;
+use BumbleDocGen\LanguageHandler\Php\Parser\Entity\ClassEntity;
+use BumbleDocGen\LanguageHandler\Php\Parser\Entity\ClassEntityCollection;
+use BumbleDocGen\LanguageHandler\Php\Parser\Entity\Exception\ReflectionException;
+use BumbleDocGen\LanguageHandler\Php\Parser\Entity\MethodEntity;
+use DI\DependencyException;
+use DI\NotFoundException;
+
+final class TemplateGenerator
+{
+    public function __construct(
+        private ProviderInterface $aiHandler,
+    ) {
+    }
+
+    private function getTemplateFromGPT(
```

Comment:
This console command has some issues with poor quality results regardless of the model. It seems to me that it makes sense to use similar logic with the readme file, since it is also a kind of readme.. only without information about installing the library and similar things

I think this can be done in a separate task

Reply:
Yeah this hasn't been updated since the Hackathon, I've some ideas for improvement, and I agree with using similar logic to the readme generator.